src/app/ui/dataselector.js

```diff
@@ -0,0 +1,48 @@
+'use client';
+
+import { useState, useContext, useEffect } from 'react';
+import Calendar from 'react-calendar';
```

Comment:
Looks like this also needs `import 'react-calendar/dist/Calendar.css';` for styling.